packages/compass-crud/src/components/bulk-update-dialog.tsx

```diff
@@ -0,0 +1,306 @@
+  textAlign: 'left',
+});
+
+const updatePreviewStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[3],
+});
+
+export type BulkUpdateDialogProps = {
+  isOpen: boolean;
+  ns: string;
+  filter: BSONObject;
+  count?: number;
+  updateText: string;
+  preview: UpdatePreview;
+  syntaxError?: Error & { loc?: { index: number } };
+  serverError?: Error;
+  closeBulkUpdateDialog: () => void;
+  updateBulkUpdatePreview: (updateText: string) => void;
+  runBulkUpdate: () => void;
+};
+
+type QueryLabelProps = {
+  tooltip: string;
+  label: string;
+};
+
+const queryLabelStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+  alignItems: 'center',
+});
+
+const QueryLabel: React.FunctionComponent<QueryLabelProps> = ({
+  tooltip,
+  label,
+}) => {
+  return (
+    <div className={queryLabelStyles}>
+      <Label htmlFor="template-dropdown">{label}</Label>
+      <InfoSprinkle align="right">{tooltip}</InfoSprinkle>
+    </div>
+  );
+};
+
+export default function BulkUpdateDialog({
+  isOpen,
+  ns,
+  filter,
+  count,
+  updateText,
+  preview,
+  syntaxError,
+  serverError,
+  closeBulkUpdateDialog,
+  updateBulkUpdatePreview,
+  runBulkUpdate,
+}: BulkUpdateDialogProps) {
+  const darkMode = useDarkMode();
+
+  const [text, setText] = useState(updateText);
+
+  const previewDocuments = useMemo(() => {
+    return preview.changes.map(
+      (change) => new HadronDocument(change.after as Record<string, unknown>)
+    );
+  }, [preview]);
+
+  const onChangeText = (value: string) => {
+    setText(value);
+    updateBulkUpdatePreview(value);
+  };
+
+  const title =
+    count === undefined ? 'Update documents' : `Update documents (${count})`;
+
+  const annotations = useMemo<Annotation[]>(() => {
+    if (syntaxError?.loc?.index) {
+      return [
+        {
+          message: syntaxError.message,
+          severity: 'error',
+          from: syntaxError.loc.index,
+          to: syntaxError.loc.index,
+        },
+      ];
+    }
+
+    return [];
+  }, [syntaxError]);
+
+  return (
+    <FormModal
+      title={title}
+      subtitle={ns}
+      size="large"
+      open={isOpen}
+      onSubmit={runBulkUpdate}
+      onCancel={closeBulkUpdateDialog}
+      cancelButtonText="Close"
+      submitButtonText="Update documents"
+      submitDisabled={!!(syntaxError || serverError)}
+    >
+      <div className={columnsStyles}>
+        <div className={queryStyles}>
+          <div className={queryFieldStyles}>
+            <TextInput
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore the label can be any component, but it's weirdly typed to string
+              label={
+                <QueryLabel
+                  label="Filter"
+                  tooltip="Return to the Documents tab to edit this query."
+                />
+              }
+              disabled={true}
+              value={(toJSString(filter) ?? '').replace(/\s+/g, ' ')}
```

Comment:
If I have a filter for a string value with multiple spaces, this reduces all those to a single one, this is probably not expected, right?

![image](https://github.com/mongodb-js/compass/assets/5036933/9ad3831c-08f7-4b2b-bd44-4c34cb1210cc)

![image](https://github.com/mongodb-js/compass/assets/5036933/6727e986-e908-413d-9f5b-06d38a8c4419)

Reply:
Oops. Leftover.